src/Facility.Definition/ServiceFieldValidation.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Facility.Definition
+{
+	/// <summary>
+	/// A validation criterion of a DTO field.
+	/// </summary>
+	public class ServiceFieldValidation
+	{
+		/// <summary>
+		/// Creates a validation criterion of a DTO field
+		/// </summary>
+		public ServiceFieldValidation(ServiceAttributeInfo attributeInfo)
+		{
+			foreach (var parameterInfo in attributeInfo.Parameters)
+			{
+				switch (parameterInfo.Name)
+				{
+					case "length":
+						LengthRange = GetRange(parameterInfo);
+						break;
+					case "count":
+						CountRange = GetRange(parameterInfo);
+						break;
+					case "value":
+						ValueRange = GetRange(parameterInfo);
+						break;
+					case "regex":
+						RegularExpression = new Regex(parameterInfo.Value);
+						break;
+					default:
+						parameterInfo.AddValidationError(ServiceDefinitionUtility.CreateUnexpectedAttributeParameterError(parameterInfo.Name, parameterInfo));
+						break;
+				}
+			}
+		}
+
+		public ServiceFieldValidationRange? CountRange { get; }
+		public ServiceFieldValidationRange? ValueRange { get; }
+		public ServiceFieldValidationRange? LengthRange { get; }
+		public Regex? RegularExpression { get; }
```

Comment:
RegexPattern seemed redundant to me, but if you prefer it over this I'm happy to change it.